src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -220,6 +258,150 @@ private void triggerCheckpoint(String checkpointIdentifier) throws FlinkExceptio
+
+        private final ReaderGroup readerGroup;
+
+        public ReaderGroupNameGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getGroupName();
+        }
+    }
+
+    /**
+     * Gauge for getting the scope name information from reader group.
+     */
+    private static class ScopeNameGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+
+        public ScopeNameGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getScope();
+        }
+    }
+
+    /**
+     * Gauge for getting online readers information from reader group.
+     */
+    private static class OnlineReadersGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String separator = ",";
+
+        public OnlineReadersGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getOnlineReaders().stream().collect(Collectors.joining(separator));
+        }
+    }
+
+    /**
+     * Gauge for getting stream name information from reader group.
+     */
+    private static class StreamNamesGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String separator = ",";
+
+        public StreamNamesGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getStreamNames().stream().collect(Collectors.joining(separator));
+        }
+    }
+
+    /**
+     * Gauge for getting position information of each segment of a stream from reader group.
+     */
+    private static class SegmentPositionsGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+        private final String stream;
+
+        public SegmentPositionsGauge(ReaderGroup readerGroup, String stream) {
+            this.readerGroup = readerGroup;
+            this.stream = stream;
+        }
+
+        @Override
+        public String getValue() {
+            StringBuilder builder = new StringBuilder();
+            builder.append("stream=").append(stream).append(", segments={");
+            Map<Stream, StreamCut> streamCuts = readerGroup.getStreamCuts();
+            Optional<Map.Entry<Stream, StreamCut>> optionalStreamCutEntry =
+                    streamCuts.entrySet().stream().filter(e -> e.getKey().getStreamName().equals(stream)).findFirst();
+            if (optionalStreamCutEntry.isPresent()) {
+                optionalStreamCutEntry.get().getValue().asImpl().getPositions().entrySet().stream().forEach(entry -> {
```

Comment:
I think it is pretty clear at this point that we need one or more string representations of a stream cut. In this particular case, we should be using a `toString` call to get it rather than doing this hack. @shrids we need to enable such call on the stream cut API.

Reply:
@shrids That's an option, but why not have a `toString` method for metrics and logging purposes? It a pretty natural way of getting it.

Reply:
Sure, will fix it as part of https://github.com/pravega/pravega/issues/2490

Reply:
@fpj @vijikarthi 
The current implementation of StreamCut already implements a `toString()` method.
i.e. if we have `StreamCut sc` which has two segment with ` segmentNumber/offset : 0/ 10  and 1/10`
then `sc.toString()` will generate a string  of format `scope/stream:0=10, 1=20` I believe this is sufficient for logging purposes.

Issue #2490 will ensure that there is a way of representing StreamCut  in Base64 / String which can be easily passed as a program parameter ; it also supports ability to construct a StreamCut back from this textual representation.